### ISSUE TITLE
Pedestrian crosswalk trajectory does not fit the path in some directions

### ISSUE THREAD

**Issue by xli149** [lgsvl/simulator#1988]
I have followed the Map annotation guide to build the pedestrian lane in order for the pedestrians to obey the traffic rules. The pedestrians' trajectory fit the pedestrian lanes on the other two direction of the crosswalk. However, they fails to fit in the lane in this direction as following. By not fitting in lane, I mean, the direction is correct, but there is a huge deviation between the pedestrian lane and the actual pedestrian trajectories no matter how I adjust the lane. Have some one encountered with this problem before?
![Screenshot from 2022-02-17 23-55-02](https://user-images.githubusercontent.com/42947993/154522519-a829ffca-f3bc-4934-bdf0-98c08d2a7bb2.png)

**Comment by EricBoiseLGSVL** [comment]
Yes, random traffic has randomization in the path and will recalculate often during runtime.  If you need precision in the path, use the api to specify the path.

**Comment by xli149** [comment]
Thank you for your explanation Eric. However, this seems still a little bit weird.  The pedestrians on the other two lanes are walking within the boundary, but tend to walk outside the boundary in this lane as following. Is there any reason for this?
![Screenshot from 2022-02-20 18-21-36](https://user-images.githubusercontent.com/42947993/154838353-1a9b3224-688d-476a-9b13-0136d7bc2f1d.png)
![Screenshot from 2022-02-20 18-21-58](https://user-images.githubusercontent.com/42947993/154838352-4cb63f97-960d-4d3f-b669-992304a97206.png)
![Screenshot from 2022-02-20 18-22-19](https://user-images.githubusercontent.com/42947993/154838354-c7638828-eb2b-4b1e-a768-6d6498739d02.png)
?

**Comment by EricBoiseLGSVL** [comment]
Yes it is a pathing issue with the ai agent code.  You can look to the pathing calls and see if you can change the behavior.  You can also look at the nav mesh settings.  There may be a setting that needs adjusted
